File: dynamic_listings/lua/default/developing_mechanics/match3_findverticalmatches.lua
function findVerticalMatches()
    local matchLength = 0
    local minMatchLength = 3
    local matches = {}
    for col = 1, #matrix do
        local column = matrix[col]
        local lastMatchingTile = nil
        for row = 1, #column do
            local currentTile = matrix[row][column].tile
            if (currentTile == lastMatchingTile) then
                matchLength = matchLength + 1
            else
                if (matchLength >= minMatchLength) then
                    -- We need to memorize all the tiles involved in the match
                    for i = row - matchLength, row do
                        local tile = matrix[i][col]
                        table.insert(matches, tile)
                    end
                else
                    -- No matches, reset the counter and set the current tile as last matching
                    matchLength = 1
                    lastMatchingTile = currentTile
                end
            end
            -- We need to account for the bottom border corner case
            if (row == #column) then
                if (matchLength >= minMatchLength) then
                    -- We need to memorize all the tiles involved in the match
                    for i = row - matchLength, row do
                        local tile = matrix[i][col]
                        table.insert(matches, tile)
                    end
                end
            end
        end
    end
    return matches
end

File: dynamic_listings/python/default/developing_mechanics/match3_findverticalmatches.py
def find_vertical_matches():
    match_length = 0
    min_match_length = 3
    for column in matrix:
        last_matching_tile = None
        for row in column:
            current_tile = matrix[row][column].tile
            if current_tile == last_matching_tile:
                match_length += 1
            else:
                if match_length >= min_match_length:
                    # We need to memorize all the tiles involved in the match
                    for r in range(row-match_length, column+1):
                        tile = matrix[r][column]
                        memorize(tile)
                else:
                    # No matches, reset the counter and set the current tile as last matching
                    match_length = 1
                    last_matching_tile = current_tile
            # We need to account for the bottom border corner case
            if row == len(matrix[column]):
                if match_length >= min_match_length:
                    # We need to memorize all the tiles involved in the match
                    for r in range(row-match_length, column+1):
                        tile = matrix[r][column]
                        memorize(tile)
